### PR TITLE
up to 20x speedup on dataset operations

### PR DESCRIPTION
**Pull request by jtwinrelevanceai**

### CODE REVIEW

File: relevanceai/operations_new/ops_run.py [hunk 1]
```diff
@@ -2,16 +2,302 @@
 Base class for base.py to inherit.
 All functions related to running operations on datasets.
 """
-import threading
+from re import L
```
Comment (boba-and-beer) [inline relevanceai/operations_new/ops_run.py:5]: Delete this

File: relevanceai/operations_new/ops_run.py [hunk 2]
```diff
@@ -2,16 +2,304 @@
...
+                overflow = pool
+
+        if overflow:
+            self.tq.put(overflow)
+            with self.lock:
+                self.pull_bar.update(len(overflow))
+
+    def _update(self):
+        while self.update_bar.n < self.ndocs:
+            try:
```
Comment (boba-and-beer) [inline relevanceai/operations_new/ops_run.py:157]: you can just use `self.tq.get(block=True)`

File: relevanceai/operations_new/sentiment/ops.py [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
 from relevanceai.operations_new.sentiment.transform import SentimentTransform
 from relevanceai.operations_new.ops_base import OperationAPIBase
 
 
 class SentimentOps(SentimentTransform, OperationAPIBase):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+    @staticmethod
+    def _get_filters(text_fields: List[str], model_name: str) -> List[str]:
```
Comment (boba-and-beer) [inline relevanceai/operations_new/sentiment/ops.py:14]: Add to base plz

File: relevanceai/operations_new/ops_run.py [hunk 4]
```diff
@@ -2,16 +2,344 @@
...
+                batch.append(document)
+
+        elif queue_id == "push":
+            queue = self.pq
+            timeout = 1
+
+            # Calculate optimal batch size
+            if self.push_batch_size is None:
+                sample_document = queue.get(timeout=timeout)
+                self.push_batch_size = self._get_optimal_push_batch_size(
```
[the remark this answers is not in the document]
Comment (jtwinrelevanceai) [reply relevanceai/operations_new/ops_run.py]: it is

**Review by boba-and-beer — APPROVED** [review]

**Pull request merged**